spark/src/main/scala/geotrellis/spark/clip/ClipToGrid.scala

```diff
@@ -16,191 +16,111 @@
 
 package geotrellis.spark.clip
 
-import geotrellis.raster.GridBounds
 import geotrellis.spark._
 import geotrellis.spark.tiling._
-import geotrellis.util._
 import geotrellis.vector._
 
-import com.vividsolutions.jts.geom.prep.{PreparedGeometry, PreparedGeometryFactory}
+import com.vividsolutions.jts.geom.prep.PreparedGeometryFactory
 import org.apache.spark.rdd._
+import scala.util.Try
 
 object ClipToGrid {
-  /** Trait which contains methods to be used in determining
-    * the most optimal way to clip geometries and features
-    * for ClipToGrid methods.
-    */
-  trait Predicates {
-    /** Returns true if the feature geometry covers the passed in extent */
-    def covers(e: Extent): Boolean
-    /** Returns true if the feature geometry is covered bythe passed in extent */
-    def coveredBy(e: Extent): Boolean
-  }
-
-  /** Clips a feature to the given extent, that uses the given predicates
-    * to avoid doing intersections where unnecessary.
-    */
-  def clipFeatureToExtent[G <: Geometry, D](
-    e: Extent,
-    feature: Feature[G, D],
-    preds: Predicates
-  ): Option[Feature[Geometry, D]] =
-    if(preds.covers(e)) { Some(Feature(e, feature.data)) }
-    else if(preds.coveredBy(e)) { Some(feature) }
-    else {
-      feature.geom.intersection(e).toGeometry.map { g =>
-        Feature(g, feature.data)
-      }
-    }
 
   /** Clip each geometry in the RDD to the set of SpatialKeys
     * which intersect it, where the SpatialKeys map to the
     * given [[LayoutDefinition]].
     */
-  def apply[G <: Geometry](
-    rdd: RDD[G],
-    layout: LayoutDefinition
-  )(implicit d: DummyImplicit): RDD[(SpatialKey, Geometry)] =
-    apply[G, Unit](rdd.map(Feature(_, ())), layout)
-      .mapValues(_.geom)
+  def geometries[G <: Geometry](layout: LayoutDefinition, rdd: RDD[G]): RDD[(SpatialKey, Geometry)] =
```

Comment:
I'm completely willing to reinstate overloaded `apply`s. I originally switched the names because the overloads were interfering with type resolution.

These names (`.geometries`, and `.features`) are also in the spirit of the new, more descriptive `MapKeyTransform` method aliases. They're a suggestion for a way forward.

Reply:
We use overloaded applys across the library, especially in this pattern of object methods -> method extensions. I get that you don't like that pattern, but changing my new code to your preferred pattern when it is not established at the project level that we will start moving to not using applys is irksome.

Reply:
No need to be irked, this stuff is never personal. As I said above I'm totally willing to use `apply` if that's deemed best. PRs are always suggestions.

Anyway, my motivation for avoiding `apply` was multifold:

- Team belief: `MapKeyTransform`'s overloaded `apply`s were mysterious to the point of warranting more descriptive aliases (i.e. good names can signal intent better).
- Practical trouble: I was having trouble getting the code to compile with the overloads.
- Team precedent: Since the `MapKeyTransform` stuff, I'm not the first to introduce a lone (i.e. bag-o'-functions only) companion object that included only descriptive names and not overloaded `apply`s.
- Personal belief: an `apply` returning a type other than its parent type is always non-mathematical.

Would you like them switched back to `apply`?

Reply:
I think we should avoid doing code changes to other people's code if it's a matter of switching from a well-established preference to a personal one, even if the personal one can be elevated to a prefered one in the future but that is not yet happend. Switching back undoes this behavior, so please switch it back.

As for moving forward, I understand your motivations. And if the project wants to set this as a 
standard, we should communicate that clearly. Your bullet points shouldn't go into a PR that changes a recent code addition, but perhaps into a GitHub issue that argues for this change in a 2.0, which we would apply across the board and then keep to moving forward.

It's possible to be irked when things are not personal; I don't think you're making this change out of anything but pure intentions. However spending cycles on debating these types of finer points is something I'd like to avoid. Which is why we should either set the standard or avoid microchanges to other's work.

Reply:
Cool, `apply` it is.

And just to be as clear as possible, the primary motivation for the rewrite was that the original form of `ClipToGrid` didn't quite match VectorPipe's use-case. I didn't discover that until checking the new code directly to figure out what the integration path would be. I should've reviewed the original PR - apologies for not having done that.

Reply:
Ok, thanks.

Let's figure out how the original code doesn't fit the use case - I had done some thinking around how it would, and so VectorPipe could rely on the code but also do whatever it wanted, like log errors in it's own specific way. What did I miss that doesn't allow VP to use things? Is it an aesthetic use case (e.g. not wanting to use Predicates in the Clip methods though it solves it) or is VP blocked from using this code for it's use case for some ohter reason?